manticore/core/smtlib/expression.py

```diff
@@ -730,8 +731,10 @@ def __add__(self, other):
         new_arr = ArrayProxy(ArrayVariable(self.index_bits, self.index_max + len(other), self.value_bits, 'concatenation{}'.format(uuid.uuid1())))
         for index in range(self.index_max):
             new_arr[index] = simplify(self[index])
+        _concrete_cache = new_arr._concrete_cache
         for index in range(len(other)):
             new_arr[index + self.index_max] = simplify(other[index])
+        new_arr._concrete_cache.update(_concrete_cache)
```

Comment:
Are those fine in the end?

It seems as if it is sth like `d = A.dict; <some operations>; A.dict.update(d)` (`d` is not a copy of `A.dict`, just a reference/pointer to it).

Reply:
The idea is that  you have two 5 bytes long arrays: xAxxx  xxBxx
1s has a concrete value in pos 1: A 
2nd has a concrete value in pos 2: B

The concatenation makes a fresh empty concrete cache dict.
It _should_ be populated with concrets from the 1st array and updated from something from the second array.

code is wrong. checking.